src/main/java/com/google/ehub/servlets/ItemPageServlet.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.ehub.data.CommentData;
+import com.google.ehub.data.CommentDataManager;
+import com.google.ehub.data.EntertainmentItem;
+import com.google.ehub.data.EntertainmentItemDatastore;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that compiles the data for the item page*/
+@WebServlet("/itempagedata")
+public class ItemPageServlet extends HttpServlet {
+  long itemId;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    itemId = Long.parseLong(request.getParameter("itemId").toString());
+    CommentDataManager commentObject = new CommentDataManager();
+    List<CommentData> comments = commentObject.retrieveComments(itemId);
+    System.out.println("doGet: " + comments);
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
I've implemented this structure more or less now but the problem I run into is that I need the ItemId for the the doPost as well because the CommentData class' addItemComment method has the ItemId as a parameter so that I can link the comments to the entity.

Reply:
Where are you calling `addItemComment`?

If you need the itemId in `doPost` you can pass it as a key/value in your `HttpServletReqest request` parameter. That way, you don't rely on the field `itemId` that is set in doGet, which could be stale. This ensures that the itemId is always the latest one we care about.

Reply:
Noted